contracts/integrations/beefy/BeefyDepository.sol

```diff
@@ -0,0 +1,242 @@
+    error NotApproved(uint256 allowance, uint256 amount);
+    error NotController(address caller);
+    error NotContractAddress(address addr);
+    error UnsupportedAsset(address asset);
+    error RedeemableSoftCapHit(uint256 softcap, uint256 totalRedeemable);
+    error TokenTransferFail(address token, address from, address to);
+
+    ///////////////////////////////////////////////////////////////////
+    ///                         Events
+    ///////////////////////////////////////////////////////////////////
+    event Deposited(
+        address indexed caller,
+        uint256 assets,
+        uint256 redeemable,
+        uint256 shares
+    );
+    event Withdrawn(
+        address indexed caller,
+        uint256 assets,
+        uint256 redeemable,
+        uint256 shares
+    );
+    event Redeemed(
+        address indexed caller,
+        uint256 assets,
+        uint256 redeemable,
+        uint256 shares
+    );
+    event RedeemableSoftCapUpdated(address indexed caller, uint256 newSoftCap);
+
+    /// @notice Constructor
+    /// @param _vaultWrapper the address of the Beefy vault wrapper
+    /// @param _controller the address of the UXDController
+    function initialize(address _vaultWrapper, address _controller)
+        external
+        virtual
+        initializer
+    {
+        __UUPSUpgradeable_init();
+        __ReentrancyGuard_init();
+        __Ownable_init();
+
+        if (!_vaultWrapper.isContract()) {
+            revert NotContractAddress(_vaultWrapper);
+        }
+        if (!_controller.isContract()) {
+            revert NotContractAddress(_controller);
+        }
+        vaultWrapper = BeefyWrapper(_vaultWrapper);
+        controller = IUXDController(_controller);
+        assetToken = address(BeefyVaultV7(vaultWrapper.vault()).want());
+        redeemable = address(controller.redeemable());
+    }
+
+    /// @dev restirct access to controller
+    modifier onlyController() {
+        if (msg.sender != address(controller)) {
+            revert NotController(msg.sender);
+        }
+        _;
+    }
+
+    /// @notice Sets the redeemable soft cap
+    /// @dev Can only be called by owner
+    /// @param softCap The new redeemable soft cap
+    function setRedeemableSoftCap(uint256 softCap) external onlyOwner {
+        redeemableSoftCap = softCap;
+        emit RedeemableSoftCapUpdated(msg.sender, softCap);
+    }
+
+    /// @notice Deposits assets
+    /// @param assetAmount The amount of assets to deposit in assetToken.decimals()
+    /// @return redeemableAmount the corresponding amount of redeemable for asset deposited
+    function deposit(address asset, uint256 assetAmount)
+        external
+        onlyController
+        returns (uint256)
+    {
+        if (asset != assetToken) {
+            revert UnsupportedAsset(asset);
+        }
+        netAssetDeposits += assetAmount;
+        IERC20(assetToken).approve(address(BeefyVaultV7(vaultWrapper.vault()).want()), assetAmount);
```

Comment:
I think here the approval should be fore `vaultWrapper.vault()` (The first parameter to `approve()`.

Reply:
Should it be the token deposited (if so it's currently correct) or the destination address? I don't have much autocompletion or advices in my IDE